category_encoders/hashing.py

```diff
@@ -238,18 +161,64 @@ def _transform_single_cpu(self, X, override_return_df=False):
         if not list(self.cols):
             return X
 
-        X = self.hashing_trick(X, hashing_method=self.hash_method, N=self.n_components, cols=self.cols)
-
-        if self.drop_invariant:
-            X = X.drop(columns=self.invariant_cols)
+        X = self.hashing_trick(
+            X, 
+            hashing_method=self.hash_method,
+            N=self.n_components, 
+            cols=self.cols, 
+        )
+        
+        return X
 
-        if self.return_df or override_return_df:
-            return X
-        else:
-            return X.to_numpy()
+    @staticmethod
+    def hash_chunk(shm_result, np_df, N, shm_offset):
+        for i, row in enumerate(np_df):
+            for val in row:
+                if val is not None:
+                    hasher = hashlib.md5()
+                    hasher.update(bytes(str(val), 'utf-8'))
+                    column_index = int.from_bytes(hasher.digest(), byteorder='big') % N
+                    row_index = (shm_offset + i)*N
+                    shm_index = row_index + column_index
+                    shm_result[shm_index] += 1
+
+    def hashing_trick_with_np_parallel(self, df, N):
+        np_df = df.to_numpy()
+        shm_result = multiprocessing.RawArray(HashingEncoder.default_int_np_array.dtype.char, len(df)*N)
+
+        n_process = []
+        chunk_size = int(len(np_df)/self.max_process)
+        ctx = multiprocessing.get_context(self.process_creation_method)
+        for i in range(0, self.max_process-1):
```

Comment:
splitting this way is not very elegant if the last chunk has to be treated separately.
using numpy array split could be helpful: https://stackoverflow.com/a/75981560.

Wouldn't it be easier if the `hash_chunk` function would hash a chunk and return an array. Then it wouldn't need the `shm_result` and `shm_offset` parameters (what does shm stand for btw?). Then you'd just concatenate all the chunks in the end?

Reply:
`shm` stands for "shared_memory" - will update the variable name 👍 

> Wouldn't it be easier if the hash_chunk function would hash a chunk and return an array. Then it wouldn't need the shm_result and shm_offset parameters (what does shm stand for btw?). Then you'd just concatenate all the chunks in the end?

That's a very good point, I will try that